financial-templates-lib/clients/ExpiringMultiPartyClient.js

```diff
@@ -90,32 +93,49 @@ class ExpiringMultiPartyClient {
     return this.lastUpdateTimestamp;
   }
 
-  async update() {
-    // Since this function can have a relatively long run-time, we should be safe and check contract state explicitly at the same block height.
-    // This way we won't experience errors where the contract state changes between calls.
-    this.latestBlock = await this.web3.eth.getBlockNumber();
+  // Returns web3.eth.Contract with an ERC20 token interface at
+  // the tokenCurrency address
+  async getTokenCurrency() {
+    const tokenCurrencyAddress = await this.emp.methods.tokenCurrency().call({}, this.latestBlock);
+    const tokenCurrency = new this.web3.eth.Contract(Token.abi, tokenCurrencyAddress);
```

Comment:
For consistency you should pass in `Token.abi` by construction of this class instead of importing the `ExpandedERC20` artifact at the top of the file. This is what we do with the EMP contract.

We might want to change this in a future refactor but I would prefer consistency right now.

Reply:
Would you like this in this PR or another PR? Adding this change will result in a lot off diffs spread out across the test files and any other classes that encapsulates the `ExpiringMultiPartyClient`

![Screenshot-2020-08-19-11:48:46](https://user-images.githubusercontent.com/9071748/90588376-d180d900-e21e-11ea-9c0f-9e2ccf733124.png)

Reply:
I feel that this logic should not be placed in the EMP client in any case. [This](https://github.com/UMAprotocol/protocol/pull/1855/files#r471953828) comment references how we should do it ideally. We don't want to start muddling these clients with async functions if possible.

Reply:
> Would you like this in this PR or another PR? Adding this change will result in a lot off diffs spread out across the test files and any other classes that encapsulates the ExpiringMultiPartyClient

Agreed with Chris that constructing and returning Token here is redundant anyways, but if it were not I would rather be consistent and then fix the construction object in a future PR.